compiler/src/iree/compiler/API/Internal/IREEGPUDialectCAPI.cpp

```diff
@@ -411,6 +413,11 @@ ireeHALExecutableTargetAttrGetGPUTargetInfo(MlirAttribute attr) {
       wgpAttr.getMaxThreadCountPerWorkgroup();
   targetInfo.maxWorkgroupMemoryBytes = wgpAttr.getMaxWorkgroupMemoryBytes();
 
+  mlir::iree_compiler::IREE::GPU::TargetChipAttr chipAttr = gpuTargetAttr.getChip();
+
+  targetInfo.wgpCount = chipAttr ? chipAttr.getWgpCount() : 0;
+  targetInfo.simdsPerWgp = wgpAttr.getSimdsPerWgp().value_or(0);
```

Comment:
With `ireeGPUTargetInfo targetInfo = {};` above, the default are already zero-initialized, so we can branch on chipAttr:


```suggestion
  if (mlir::iree_compiler::IREE::GPU::TargetChipAttr chipAttr = gpuTargetAttr.getChip()) {
    targetInfo.wgpCount = chipAttr.getWgpCount();
  }
  targetInfo.simdsPerWgp = wgpAttr.getSimdsPerWgp();
```

Reply:
I have a Q here: simdsPerWgp is not dependent on on TargetChipAttr.

Reply:
[f4340d5](https://github.com/iree-org/iree/pull/22527/commits/f4340d5c298c3d3f15b52e8b130f4c9e0e2f0011)